demangle/src/lib.rs

```diff
@@ -181,23 +184,65 @@ fn try_demangle_objcpp(ident: &str, opts: DemangleOptions) -> Option<String> {
     }
 }
 
-/// Allows to demangle potentially mangled names.
+/// An extension trait on `Name` for demangling names.
+///
+/// See the [module level documentation] for a list of supported languages.
 ///
-/// Non-mangled names are largely ignored and language detection will not
-/// return a language. Upon formatting, the symbol is automatically demangled
-/// (without arguments).
+/// [module level documentation]: index.html
 pub trait Demangle {
     /// Infers the language of a mangled name.
     ///
-    /// In case the symbol is not mangled or not one of the supported languages
-    /// the return value will be `None`. If the language of the symbol was
-    /// specified explicitly, this is returned instead.
+    /// In case the symbol is not mangled or its language is unknown, the return value will be
+    /// `Language::Unknown`. If the language of the symbol was specified explicitly, this is
+    /// returned instead. For a list of supported languages, see the [module level documentation].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use symbolic::common::{Name, Language};
+    /// use symbolic::demangle::{Demangle, DemangleOptions};
+    ///
+    /// assert_eq!(Name::new("_ZN3foo3barEv").detect_language(), Language::Cpp);
+    /// assert_eq!(Name::new("unknown").detect_language(), Language::Unknown);
+    /// ```
+    ///
+    /// [module level documentation]: index.html
     fn detect_language(&self) -> Language;
 
     /// Demangles the name with the given options.
+    ///
+    /// Returns `None` in one of the following cases:
+    ///  1. The language cannot be detected.
+    ///  2. The language cannot be demangled.
```

Comment:
the language?

Reply:
probably better to say "is not supported"